pypesto/sample/variational_inference.py

```diff
@@ -0,0 +1,153 @@
+import logging
+from time import process_time
+from typing import List, Optional, Union
+
+import numpy as np
+from scipy import stats
+
+from ..problem import Problem
+from ..result import Result
+
+# from ..store import autosave
+from .pymc import PymcSampler
+from .util import bound_n_samples_from_env
+
+logger = logging.getLogger(__name__)
+
+
+def variational_fit(
+    problem: Problem,
+    n_iterations: int,
+    method: str = 'advi',
+    # n_samples: Optional[int] = None,
+    random_seed: Optional[int] = None,
+    start_sigma: Optional[dict[str, np.ndarray]] = None,
+    x0: Union[np.ndarray, List[np.ndarray]] = None,
+    result: Result = None,
+    # filename: Union[str, Callable, None] = None,
+    # overwrite: bool = False,
+    **kwargs,
+) -> Result:
+    """
+    Call to do parameter sampling.
+
+    Parameters
+    ----------
+    problem:
+        The problem to be solved. If None is provided, a
+        :class:`pypesto.AdaptiveMetropolisSampler` is used.
+    n_iterations:
+        Number of iterations for the optimization.
+    method: str or :class:`Inference`
```

Comment:
Unless we link our documentation to pymc, we have no class inference I think 🤔  Should also be indicated in the type of `method`.